### ISSUE TITLE
Quest: Map crossing=island

### ISSUE THREAD

**Issue by rugk** [westnordost/StreetComplete#495]
Continuing the discussion [from here](https://github.com/westnordost/StreetComplete/pull/432#issuecomment-320043587).

As always the implementation is great, thumbs up to @ENT8R, but it would be greater with that 🏝. :wink: 
Especially as here, in Germany and Austria these are not that uncommon.

My suggestion would be to just add a small message box below the three images, which says:
> [ ] This crossing has a traffic isle in the middle.

As that "property" can, of course, apply to all these three images, that is reasonable, IMHO. And it does not take much vertical space away…
Downside: You have no image there. But if you want to be sure, maybe display the image (similar to how you show a popup for livingstreet or so) when the user first clicks on taht checkbox and let him/her confirm the decision. Afterwards, this is hopefully not necessary anymore as they "learned" what a traffic island is. (and that it is not :desert_island: :wink:)

Alternatively, try to get it into "Other answers" somehow? (Don't know how yet, however, because you also need the "initial" answer to the main quest.)

/cc @Etua @krzyk

**Comment by westnordost** [comment]
I think we have established by now that the traffic island is a separate property and thus should not be answered in the same quest.

**Comment by rugk** [comment]
Hmm, okay… However with StreetComplete we cannot change the properties/elements.

So e.g. this is what we want (`----` is the crossing):
```
|             |
| ---- o ---- |
   ^       ^
```

There are two quests and we do not need `traffic-island`. However, what is, if it actually is mapped "incorrectly":
```
|             |
| ----------- |
       ^
```

There one quest is asked. So now, the StreetComplete user should have the possibility to indicate there is an island. A(nother) mapper could later still change the crossing to properly modify it.

For doing so, we could:
* use crossing=traffic-island, as requested
* OR: add a note on OSM stating the same issue
* Or, theoretically, StreetComplete could split the thing in the middle by itself. But I know, you likely won't implement this.

As for the UI, the proposal remains the same.

Now: Is it worth it?
That's where we need data… How common is the "incorrect" case? If it is popular, it makes sense to include that checkbox.
I have just no idea how to get data on this, as we, of course, need historical data…

**Comment by westnordost** [comment]
I don't understand your example.

**Comment by rugk** [comment]
The first one is with an island. The second is not. Now clear? (Was just some try to make it visual…🙄)

**Comment by westnordost** [comment]
No

**Comment by rugk** [comment]
Ah, I think I misunderstood your comment.
> thus should not be answered in the same quest

Okay, I don't care and I am fine with using another quest for that. Then take this issue as such.
But why not use a simple checkbox in the usual crossing quest simplify the whole thing?

The problem with a separate quest is, how does one/StreetComplete mark a crossing as "not an island"? AFAIK this is not possible and thus such crossings would always get the repeated quest: "Is there an island?"
So when one can answer whether a crossing has a :vertical_traffic_light: then one can also see whether there is a traffic island in the middle… Why not combine that?
Of course, sometimes traffic island are mapped by splitting the crossing into two parts anyway, but this is something the user can see on the map.

Also AFAIK it is not "a separate property", it still belongs into the crossing=* key…

**Comment by westnordost** [comment]
Because I consider the tagging scheme as it is documented so problematic that I do not want to use it. Rather than inventing a new scheme for that use case to tackle the very problem you mention - yes, you identified the problem correctly - I rather do not have it at all.

Frankly, I just want to work on and maintain this app. I don't want to additionally take care of and embroil myself in tagging discussions. If a tagging scheme is not usable for this app the way it is (documented), then I leave it.

Basically, see the discussion in #432.

P.S: This has only marginally something to do with this, but did you read http://blog.imagico.de/social-engineering-in-openstreetmap/ ? I found it an interesting read.

**Comment by westnordost** [comment]
(Closed because you correctly identified that with the current tagging scheme, there is no way to distinguish between "no traffic island" and "traffic island hasn't been identified yet")

**Comment by rugk** [comment]
Okay, I see your point although I'd still advocate for a new quest at least. (And yeah, I've read most of the PR already.) In any case, I see you see a root problem, so to move (OSM as a whole) forward, this issue should be addressed. So to tackle the problem at the root, I (re)raise a discussion [in the wiki](https://wiki.openstreetmap.org/wiki/Talk:Key:crossing#Island_.2Aor.2A_Traffic_signals_.3F).
